docs/feature_combo.md

```diff
@@ -71,26 +92,118 @@ void process_combo_event(uint16_t combo_index, bool pressed) {
+
+## Per Combo Timing and Holding
+Instead of using a blanket `COMBO_TERM` window for every combo, per-combo timings can be set. In order to use this feature the following config options and functions need to be defined. Coming up with useful timings and configuration is left as an exercise for the reader.
+
+| Config Flag                 | Function                                                  | Description                                                                                          |
+|-----------------------------|-----------------------------------------------------------|------------------------------------------------------------------------------------------------------|
+| `COMBO_MUST_HOLD_PER_COMBO` | bool get_combo_must_hold(uint16_t index, combo_t \*combo) | Controls if a given combo should fire immediately on tap or if it needs to be held. (default: False) |
+| `COMBO_TERM_PER_COMBO`      | uint16_t get_combo_term(uint16_t index, combo_t \*combo)  | Optional per-combo timeout window. (default: COMBO_TERM)                                             |
+
+If you set either of these options but would like to keep the Modifier Combo settings, you have to define those yourself in the function.
+
+Examples:
+```c
+uint16_t get_combo_term(uint16_t index, combo_t *combo) {
+    // decide by combo->keycode
+    if (KEYCODE_IS_MOD(combo->keycode)) return COMBO_MOD_TERM; // you have to config this yourself if you're using COMBO_TERM_PER_COMBO
+
+    switch (combo->keycode) {
+        case KC_X:
+            return 50;
+    }
+
+    // or with combo index, i.e. its name from enum.
+    switch (index) {
+        case COMBO_NAME_HERE:
+            return 9001;
+    }
+
+    // And if you're feeling adventurous, you can even decide by the keys in the chord,
+    // i.e. the exact array of keys you defined for the combo.
+    // This can be useful if your combos have a common key and you wan't to apply the
+    // same combo term for all of them.
+    if (combo->keys[0] == KC_ENTER) { // if first key in the array is KC_ENTER
+        return 150;
+    }
+
+    return COMBO_TERM;
+}
+
+bool get_combo_must_hold(uint16_t index, combo_t *combo) {
+    // yet again, with keycode and true for Modifiers if so desired.
+    if (KEYCODE_IS_MOD(combo->keycode)) return true;
+
+    // or index/name
+    switch (index) {
+        case COMBO_NAME_HERE:
+            return true;
+    }
+
+    return false;
+}
+```
+
+## Variable Length Combos
+If you leave `COMBO_COUNT` undefined in `config.h`, it allows you to programmatically declare the size of the Combo data structure and avoid updating COMBO_COUNT. Instead a variable called `COMBO_LEN` has to be set. It can be set with something similar to the following in `keymap.c`: `uint16_t COMBO_LEN = sizeof(key_combos) / sizeof(key_combos[0]);` or by adding `COMBO_LENGTH` as the *last* entry in the combo enum and then `uint16_t COMBO_LEN = COMBO_LENGTH;` as such:
```

Comment:
```suggestion
If you leave `COMBO_COUNT` undefined in `config.h`, it allows you to programmatically declare the size of the Combo data structure and avoid updating `COMBO_COUNT`. Instead a variable called `COMBO_LEN` has to be set. It can be set with something similar to the following in `keymap.c`: `uint16_t COMBO_LEN = sizeof(key_combos) / sizeof(key_combos[0]);` or by adding `COMBO_LENGTH` as the *last* entry in the combo enum and then `uint16_t COMBO_LEN = COMBO_LENGTH;` as such:
```